x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -6,154 +6,252 @@
+         minLon: _.round(mbBounds.getWest(), DECIMAL_DEGREES_PRECISION),
+         minLat: _.round(mbBounds.getSouth(), DECIMAL_DEGREES_PRECISION),
+         maxLon: _.round(mbBounds.getEast(), DECIMAL_DEGREES_PRECISION),
+         maxLat: _.round(mbBounds.getNorth(), DECIMAL_DEGREES_PRECISION)
+       }
+     };
+   }
+
+   componentDidMount() {
+     this._initializeMap();
+     this._isMounted = true;
+   }
+
+   componentWillUnmount() {
+     this._isMounted = false;
+     if (this._checker) {
+       this._checker.destroy();
+     }
+     if (this._mbMap) {
+       this._mbMap.remove();
+       this._mbMap.off('mousemove', this._onMouseMove);
+       this._mbMap = null;
+       this._tooltipContainer = null;
+     }
+     this.props.onMapDestroyed();
+   }
+
+   async _initializeMap() {
+
+
+     this._mbMap = await createMbMapInstance(this.refs.mapContainer, this.props.goto ? this.props.goto.center : null);
+
+     if (!this._isMounted) {
+       return;
+     }
+     // Override mapboxgl.Map "on" and "removeLayer" methods so we can track layer listeners
+     // Tracked layer listerners are used to clean up event handlers
+     const originalMbBoxOnFunc = this._mbMap.on;
+     const originalMbBoxRemoveLayerFunc = this._mbMap.removeLayer;
+     this._mbMap.on = (...args) => {
+       // args do not identify layer so there is nothing to track
+       if (args.length <= 2) {
+         originalMbBoxOnFunc.apply(this._mbMap, args);
+         return;
+       }
+
+       const eventType = args[0];
+       const mbLayerId = args[1];
+       const handler = args[2];
+       this._addListener(eventType, mbLayerId, handler);
+
+       originalMbBoxOnFunc.apply(this._mbMap, args);
+     };
+     this._mbMap.removeLayer = (id) => {
+       this._removeListeners(id);
+       originalMbBoxRemoveLayerFunc.apply(this._mbMap, [id]);
+     };
+
+     this._initResizerChecker();
+
+     //todo :these callbacks should be removed on destroy
+     // moveend callback is debounced to avoid updating map extent state while map extent is still changing
+     // moveend is fired while the map extent is still changing in the following scenarios
+     // 1) During opening/closing of layer details panel, the EUI animation results in 8 moveend events
+     // 2) Setting map zoom and center from goto is done in 2 API calls, resulting in 2 moveend events
+     this._mbMap.on('moveend', _.debounce(() => {
+       this.props.extentChanged(this._getMapState());
+     }, 100));
+
+     const throttledSetMouseCoordinates = _.throttle(e => {
+       this.props.setMouseCoordinates({
+         lat: e.lngLat.lat,
+         lon: e.lngLat.lng
+       });
+     }, 100);
+     this._mbMap.on('mousemove', throttledSetMouseCoordinates);
+     this._mbMap.on('mouseout', () => {
+       throttledSetMouseCoordinates.cancel(); // cancel any delayed setMouseCoordinates invocations
+       this.props.clearMouseCoordinates();
+     });
+
+
+     this._mbMap.on('mousemove', this._onMouseMove);
```

Comment:
We have two methods defined for `mousemove` which is fine, just the callback `_onMouseMove` is ambiguous. How about renaming `_onMouseMove` to `_updateTooltipState`?